src/integration/billing/stripe/StripeBillingIntegration.ts

```diff
@@ -1194,102 +1155,86 @@ export default class StripeBillingIntegration extends BillingIntegration<StripeB
+    // Let's check if the STRIPE customer exists
+    const customerID:string = user?.billingData?.customerID;
+    if (!customerID) {
+      throw new Error('Unexpected situation - the customerID is NOT set');
+    }
+    let customer = await this.getStripeCustomer(customerID);
+    const userID = customer?.metadata?.['userID'];
+    if (userID !== user.id) {
+      throw new Error('Unexpected situation - the STRIPE metadata does not match');
+    }
+    // Update user data
+    const updateParams: Stripe.CustomerUpdateParams = {
+      description,
+      name,
+      email: user.email,
+      preferred_locales: [locale]
+    };
+    // Update changed data
+    customer = await this.stripe.customers.update(customerID, updateParams);
+    // Let's update the Billing Data of our customer
+    // TODO - to be clarified - The caller is here responsible for storing the changes!!!
+    user.billingData.lastChangedOn = new Date();
+    // Let's return the corresponding Billing User
+    return this.convertToBillingUser(customer, user);
   }
 
   public async deleteUser(user: User): Promise<void> {
     // Check Stripe
     await this.checkConnection();
     // const customer = await this.getCustomerByEmail(user.email);
-    const customer = await this.getStripeCustomer(user);
+    const customerID = user.billingData?.customerID;
+    const customer = await this.getStripeCustomer(customerID);
     if (customer && customer.id) {
       await this.stripe.customers.del(
         customer.id
       );
     }
   }
 
-  private async getStripeCustomer(userOrCustomerID: User | string, strictMode = true): Promise<Stripe.Customer> {
-    await this.checkConnection();
-    // Get customer
-
-    let customerID ;
-    if (typeof userOrCustomerID === 'string') {
-      customerID = userOrCustomerID;
-    } else {
-      customerID = userOrCustomerID?.billingData?.customerID;
-    }
-
+  private async getStripeCustomer(customerID: string): Promise<Stripe.Customer> {
     if (customerID) {
       try {
-        const customer: Stripe.Customer = await this.stripe.customers.retrieve(customerID) as Stripe.Customer;
-        return customer;
-      } catch (error) {
-        // ---------------------------------------------------------------------------------------
-        // This should not happen - The customerID
-        // The customerID refers to something which does not exists anymore in the STRIPE account
-        // May happen when billing settings are changed to point to a different STRIPE account
-        // ---------------------------------------------------------------------------------------
-        if (strictMode) {
+        // Gets the STRIPE Customer
+        const customer: Stripe.Customer | Stripe.DeletedCustomer = await this.stripe.customers.retrieve(customerID);
+        // Check for the deletion flag
+        const deleted = (customer.deleted) ? true : false; // ACHTUNG - STRIPE type definition is wrong!
```

Comment:
for me it works. tried assigning myself a deleted cus_ and returned a Stripe.DeletedCustomer